tests/x11/user_defined_snapshot.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Show user defined comments in grub2 menu for snapshots
+# Maintainer: Dumitru Gutu <dgutu@suse.com>
+
+use base "x11test";
+use strict;
+use testapi;
+use utils;
+
+sub y2snapper_create_snapshot() {
+    my ($self, $name, $user_data) = @_;
+    $name      //= 'grub_comment';
+    $user_data //= 'bootloader="Bootloader_Comment"';
+    # Open the 'C'reate dialog and wait until it is there
+    send_key "alt-c";
+    assert_screen 'yast2_snapper-createsnapshotdialog', 100;
+    # Fill the form and finish by pressing the 'O'k-button
+    type_string $name;
+    send_key "alt-u";    # match User data column
+    type_string $user_data;
+    save_screenshot;
+    send_key "alt-o";
+    save_screenshot;
+}
+
+sub run() {
+    my $self = shift;
+    # Start an xterm as root
+    x11_start_program("xterm");
+    wait_idle;
```

Comment:
please replace with something else. We want to avoid `wait_idle` because on some machines it is just a stupid (and very long) sleep. Better an explicit assert_screen which we have already for xterm.